### PR TITLE
Add missing LED Matrix suspend code to suspend.c

### PR DESCRIPTION
**Pull request by drashna**
## Description

Realized that I never added the suspend function calls for LED Matrix.  Whoops. 

## Types of Changes

- [x] Core
- [x] Bugfix

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: tmk_core/common/avr/suspend.c [hunk 1]
```diff
@@ -167,6 +167,10 @@ void suspend_power_down(void) {
     rgb_matrix_set_suspend_state(true);
 #    endif
 
+#    if defined(LED_MATRIX_ENABLE)
```
Comment (fauxpark) [inline tmk_core/common/avr/suspend.c:170]: Let's swap these around to match everywhere else LED/RGB Matrix related stuff is conditionally added.

File: tmk_core/common/chibios/suspend.c [hunk 2]
```diff
@@ -86,6 +86,10 @@ void suspend_power_down(void) {
 #if defined(RGB_MATRIX_ENABLE)
     rgb_matrix_set_suspend_state(true);
 #endif
+#if defined(LED_MATRIX_ENABLE)
+    led_matrix_set_suspend_state(true);
+#endif
```
Comment (fauxpark) [inline tmk_core/common/chibios/suspend.c:91]:
```suggestion
#if defined(LED_MATRIX_ENABLE)
    led_matrix_set_suspend_state(true);
#endif
#if defined(RGB_MATRIX_ENABLE)
    rgb_matrix_set_suspend_state(true);
#endif
```

File: tmk_core/common/chibios/suspend.c [hunk 3]
```diff
@@ -156,6 +160,9 @@ void suspend_wakeup_init(void) {
 #endif
 #if defined(RGB_MATRIX_ENABLE)
     rgb_matrix_set_suspend_state(false);
+#endif
+#if defined(LED_MATRIX_ENABLE)
+    led_matrix_set_suspend_state(false);
 #endif
```
Comment (fauxpark) [inline tmk_core/common/chibios/suspend.c:166]:
```suggestion
#if defined(LED_MATRIX_ENABLE)
    led_matrix_set_suspend_state(false);
#endif
#if defined(RGB_MATRIX_ENABLE)
    rgb_matrix_set_suspend_state(false);
#endif
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: tmk_core/common/avr/suspend.c [hunk 4]
```diff
@@ -218,6 +222,9 @@ void suspend_wakeup_init(void) {
 #if defined(RGBLIGHT_SLEEP) && defined(RGBLIGHT_ENABLE)
     rgblight_wakeup();
 #endif
+#if defined(LED_MATRIX_ENABLE)
```
Comment (fauxpark) [inline tmk_core/common/avr/suspend.c:225]:
```suggestion

#if defined(LED_MATRIX_ENABLE)
```

File: tmk_core/common/avr/suspend.c [hunk 5]
```diff
@@ -163,6 +163,10 @@ void suspend_power_down(void) {
     rgblight_suspend();
 #    endif
 
+#    if defined(LED_MATRIX_ENABLE)
+    led_matrix_set_suspend_state(true);
+#    endif
+
```
Comment (fauxpark) [inline tmk_core/common/avr/suspend.c:169]:
```suggestion
```

File: tmk_core/common/chibios/suspend.c [hunk 6]
```diff
@@ -83,6 +83,9 @@ void suspend_power_down(void) {
 #if defined(RGBLIGHT_SLEEP) && defined(RGBLIGHT_ENABLE)
     rgblight_suspend();
 #endif
+#if defined(LED_MATRIX_ENABLE)
```
Comment (fauxpark) [inline tmk_core/common/chibios/suspend.c:86]:
```suggestion

#if defined(LED_MATRIX_ENABLE)
```

File: tmk_core/common/chibios/suspend.c [hunk 7]
```diff
@@ -154,6 +157,9 @@ void suspend_wakeup_init(void) {
 #if defined(RGBLIGHT_SLEEP) && defined(RGBLIGHT_ENABLE)
     rgblight_wakeup();
 #endif
+#if defined(LED_MATRIX_ENABLE)
```
Comment (fauxpark) [inline tmk_core/common/chibios/suspend.c:160]:
```suggestion

#if defined(LED_MATRIX_ENABLE)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: tmk_core/common/avr/suspend.c (hunk 5 above)
Comment (drashna) [reply tmk_core/common/avr/suspend.c:169]: The spacing is consistent with how the file was. But ...

**Review by fauxpark — APPROVED** [review]

**Pull request merged**